.github/workflows/continuous-build.yml

```diff
@@ -185,35 +186,50 @@ jobs:
 
       # workaround for handling input value for auto-triggered workflows
       # https://dev.to/mrmike/github-action-handling-input-default-value-5f2g
-      - name: Set AWS region value
+      - name: Set AWS region and partition values
         id: setRegion
         run: |
           USER_INPUT=${{ github.event.inputs.region }}
           # Use user input region or fall back to "us-west-2"
           REGION=${USER_INPUT:-"us-west-2"}
           echo "::set-output name=region::$REGION"
+          echo "::set-output name=partition::aws"
+
+          if [[ ${{ matrix.testcases }} == "linux_zip_cn" ]]; then
+            echo "::set-output name=region::cn-north-1"
+            echo "::set-output name=partition::aws-cn"
+          fi
 
       - name: Configure AWS Credentials
+        if: ${{ steps.setRegion.outputs.partition == 'aws' }}
         uses: aws-actions/configure-aws-credentials@v1
         with:
           role-to-assume: ${{ secrets.TESTING_ROLE_ARN }}
           aws-region: ${{ steps.setRegion.outputs.region }}
 
+      - name: Configure AWS Credentials - China
+        if: ${{ steps.setRegion.outputs.partition == 'aws-cn' }}
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.TESTING_ROLE_ARN_CN }}
+          aws-region: cn-north-1
+
       - name: Generate trace data
         id: traceData
         run: |
           sudo chmod +x ./testing/generate_trace_data.sh
           ./testing/generate_trace_data.sh
 
       - name: Run terraform and send trace data
+        id: terraformSendData
         run: |
           cd testing/terraform
           terraform init
           terraform validate
           terraform apply -auto-approve -var-file=./testcases/${{ matrix.testcases }}.tfvars -var=aws_region=${{ steps.setRegion.outputs.region }}
 
       - name: Validate trace data
-        if: always()
+        if: steps.terraformSendData.outcome == 'success'
```

Comment:
Do we need to alert if the terraform failed to apply?

Reply:
If the terraform apply step fails, the workflow will fail and we will be notified via the email configured on the repo.
With this change, it tells the workflow to only validate trace if the terraform apply and send trace data step was successful. Otherwise this step would run for 1 minute even though we know that the validation will surely fail.